addons/purchase_requisition/tests/test_purchase_requisition.py

```diff
@@ -348,3 +348,66 @@ def test_11_alternative_po_from_po_with_requisition_id(self):
 
         po_2 = po_1.alternative_po_ids - po_1
         self.assertFalse(po_2.requisition_id, "The requisition_id should not be set in the alternative purchase order")
+
+    def test_12_alternative_po_line_different_currency(self):
+        """ Check alternative PO with different currency is compared correctly"""
+        vendor_eur = self.env['res.partner'].create({
+            'name': 'eur',
+        })
+        vendor_usd = self.env['res.partner'].create({
+            'name': 'eur',
+        })
```

Comment:
No need to make new partners (unneccesary `create`s in tests slow them down). Use the existing `self.res_partner_1` since you set the currency in the PO anyways (the same vendor can have different POs with different currencies)